src/ui/component/fileRender/view.jsx

```diff
@@ -154,25 +105,12 @@ class FileRender extends React.PureComponent<Props> {
       // @if TARGET='app'
       '3D-file': <ThreeViewer source={{ fileType, downloadPath }} theme={currentTheme} />,
       'comic-book': <ComicBookViewer source={{ fileType, downloadPath }} theme={currentTheme} />,
+      application: <AppViewer uri={uri} />,
       // @endif
 
-      application: !source.url ? null : (
-        <webview
-          ref={element => this.processSandboxRef(element)}
-          title=""
-          sandbox="allow-scripts allow-forms allow-pointer-lock"
-          src={source.url}
-          autosize="on"
-          style={{ border: 0, width: '100%', height: '100%' }}
-          useragent="Mozilla/5.0 AppleWebKit/537 Chrome/60 Safari/537"
-          enableremotemodule="false"
-          webpreferences="sandbox=true,contextIsolation=true,webviewTag=false,enableRemoteModule=false,devTools=false"
-        />
-      ),
-      video: (
-        <VideoViewer claim={claim} source={{ downloadPath, fileName }} contentType={contentType} poster={poster} />
-      ),
-      audio: <VideoViewer claim={claim} source={{ downloadPath, fileName }} contentType={contentType} />,
+      video: <VideoViewer source={streamingUrl} contentType={contentType} />,
```

Comment:
does the creation of `mediaTypes` trigger the instantiation/invocation of the individual components? or is that deferred until the element is used in some way? (I think it's the latter, but making sure)

Reply:
If download path exists and the file is completed, we should use that for the video and audio player.